src/util/exectools.ts

```
@@ -94,3 +97,39 @@ export async function updateBuildDeps(
     return false;
 }
 
+/**
+ * Attempts to shut down Bazel, first gracefully, then forcefully if necessary.
+ * @param {number} timeoutMs - The time to wait for a graceful shutdown before forcing it.
+ */
+export async function shutdownBazelHard(timeoutMs = 2000) {
+    try {
+        // Create a timeout promise
+        const timeoutPromise = new Promise((_, reject) =>
+            setTimeout(() => reject(new Error('Graceful shutdown timed out')), timeoutMs)
+        );
+
+        // Race the shutdown command against the timeout
+        await Promise.race([
+            executeCommand('bazel shutdown', getRoot()),
+            timeoutPromise
+        ]);
+
+        return;
+
+    } catch (error) {
```

if you're just catching and ignoring this error, is it redundant to even specify `new Error` and it's message up above? Maybe not for one debugging, but I'm just curious